vendor/bat-native-ads/src/bat/ads/internal/ad_events/search_result_ads/search_result_ad_event_served.cc

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/ad_events/search_result_ads/search_result_ad_event_served.h"
+
+#include "bat/ads/confirmation_type.h"
+#include "bat/ads/internal/ad_events/ad_events.h"
+#include "bat/ads/internal/ads_history/ads_history.h"
+#include "bat/ads/internal/logging.h"
+
+namespace ads {
+namespace search_result_ads {
+
+AdEventServed::AdEventServed() = default;
+
+AdEventServed::~AdEventServed() = default;
+
+void AdEventServed::FireEvent(const SearchResultAdInfo& ad) {
+  BLOG(3, "Served search result ad with uuid " << ad.uuid
+                                               << " and creative instance id "
+                                               << ad.creative_instance_id);
+
+  LogAdEvent(ad, ConfirmationType::kServed, [](const bool success) {
+    if (!success) {
+      BLOG(1, "Failed to log search result ad served event");
+      return;
+    }
+
+    BLOG(6, "Successfully logged search result ad served event");
+  });
+}
```

Comment:
do we need to add adding to history here? `history::AddSearchResultAd`

Reply:
We only add to the history when the ad is viewed. Search ads will not be served by the library for the MVP so once a view event is fired a served event is automatically fired. Thanks